docs/input-logstash.asciidoc

```diff
@@ -193,20 +168,59 @@ The certificate _SHOULD_ include identity claims about the ip address or hostnam
 One or more PEM-encoded files defining certificate authorities for use in client authentication.
-
 This setting can be used to _override_ the system trust store for verifying the SSL certificate presented by clients.
 
+[id="plugins-{type}s-{plugin}-ssl_client_authentication"]
+===== `ssl_client_authentication`
+
+* Value can be any of:
+** `none`: do not request client's certificate, or validate certificates that are presented
+** `optional`: request client's certificate, and validate it against our trust authorities _if-and-only-if_ it is presented
+** `required`: require a valid certificate from the client that is signed by a trusted certificate authority
+* Default value is `"none"`
+
+By default, the server doesn't proceed client authentication.
+This means that connections from clients are _private_ when SSL is enabled, but that this input will allow SSL connections from _any_ client.
+If you wish to configure this plugin to reject connections from untrusted hosts, you will need to configure this plugin to authenticate clients, and may also need to configure its <<plugins-{type}s-{plugin}-config-ssl-trust, source of trust>>.
+
+[id="plugins-{type}s-{plugin}-ssl_cipher_suites"]
+===== `ssl_cipher_suites`
+
+* Value type is <<array,array>>
+* Default value is `['TLS_AES_256_GCM_SHA384', 'TLS_AES_128_GCM_SHA256', 'TLS_CHACHA20_POLY1305_SHA256', 'TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384', 'TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384', 'TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256', 'TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256', 'TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256', 'TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256', 'TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384', 'TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384', 'TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256', 'TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256']`
+
+The list of cipher suites to use, listed by priorities.
+This default list applies for OpenJDK 11.0.14 and higher.
+For older JDK versions, the default list includes only suites supported by that version.
+For example, the ChaCha20 family of ciphers is not supported in older versions.
+
+[id="plugins-{type}s-{plugin}-ssl_enabled"]
+===== `ssl_enabled`
+
+* Value type is <<boolean,boolean>>
+* Default value is `true`
+
+SSL is enabled by default, which requires configuring this plugin to present its <<plugins-{type}s-{plugin}-config-ssl-identity, identity>>.
+
+You can disable SSL with `+ssl_enabled => false+`. When disabled, setting any `ssl_*` configuration causes configuration failure.
+
+[id="plugins-{type}s-{plugin}-ssl_handshake_timeout"]
+===== `ssl_handshake_timeout`
+
+* Value type is <<number,number>>
+* Default value is `10000`
+
+Time in milliseconds for SSL handshake timeout
+
 [id="plugins-{type}s-{plugin}-ssl_key"]
 ===== `ssl_key`
 
 * Value type is <<path,path>>
 * There is no default value for this setting.
 * Required when connection identity is configured with <<plugins-{type}s-{plugin}-ssl_certificate>>
-* {prohibit-ssl-disabled-effective}
```

Comment:
We need to _inline_ the shared attributes, not dereference them, such as:

~~~ diff
-* {prohibit-ssl-disabled-effective}
+* Cannot be combined with configurations that disable SSL
~~~

Reply:
Initially, I was thinking same that highlighting impossibility using with `ssl_enabled => false`.
After some thoughts, I [prefered to put them under `ssl_enabled` umbrella](https://github.com/logstash-plugins/logstash-integration-logstash/pull/3/files#diff-f4d6517546b1e35271ac597b94cf7b50702c26ffdaaae5e399e096aac9ae997fR208), telling that all `ssl_*` params cannot be used when SSL is disabled.

Reply:
When looking at a specific configuration option, I would prefer for the requirements for _that option_ to be apparent to me, all in one place.

I am fine with a clause in the `ssl_enabled` that conveys that configuring the _specifics_ of SSL is prohibited when SSL is disabled, but we should also have clear docs for each option that say when it cannot be provided.

Reply:
Additionally, these caveats were in the option bullet-points before, because they are meant to be understood as individual facts and adding them to the descriptive block makes that block harder to read.

Reply:
As we agreed offline, referenced them inline.

As a side note, I kept the alphabetical order standard and will discuss FAQ/story style (I personally support) with @karenzone when we start docs.

Reply:
Ideally, we name settings so that they naturally fall together when grouped in alphabetical order.